### PR TITLE
Fix - Calendar widget jumps when switching months

### LINKED ISSUE #39487: Calendar widget jumps when switching months
### Describe the bug

When the calendar widget is opened and you switch months, the height of the widget depends on the months selected and it jumps every other time.
If you open it from bottom between filter for the [timeseries chart like this](https://stats.metabase.com/question/16356-orders-count-grouped-by-year-and-created-at-month-of-year-filtered-by-year-is-between-2023-and-2024) you have to move mouse cursor to catch the running next month button which is pretty annoying when you want to click through several months.
Captura de ecrã 2024-03-02, às 02 42 16
Captura de ecrã 2024-03-02, às 02 42 23
If you open it from the chill mode Filter button, the Update filter button jumps.
Captura de ecrã 2024-03-02, às 02 56 23
Captura de ecrã 2024-03-02, às 02 56 32
If you open it from the query builder Filter button, the Update filter button is cropped
Captura de ecrã 2024-03-02, às 02 57 12

### To Reproduce

1. Click Between in any Filter menu
2. Click next or previous month in the calendar widget several times
3. The widget height changes depending on the months chosen

### Expected behavior

The calendar widget height should be fixed, the buttons should not move or be cropped

### Logs

_No response_

### Information about your Metabase installation

```JSON
stats.metabase.com
```

### Severity

Just annoying

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by kamilmielnik**
Fixes #39487

### Description

Fixes layout shift in calendar when using previous/next buttons by adding `mih` prop to the `DatePicker` component.

### How to verify

Stress test: https://github.com/metabase/metabase/actions/runs/10039093680

Dimensions:
- Filter types:
    - single date picker ("Before"/"After"/"On" filters)
    - date range picker ("Between" filter)
    - height should change if other filers are selected ("All time"/"Previous"/"Next"/"Current").
- Usages:
    - time filter
    - filter modal
    - filter pills
    - filter drill
    - notebook editor filter

#### Before

https://github.com/user-attachments/assets/a5a98a41-f89c-4667-b31e-0a9940f5c253

#### After

https://github.com/user-attachments/assets/df1e4fe0-fd3c-47ea-a21f-de483369325e

### CODE REVIEW

File: e2e/test/scenarios/question-reproductions/reproductions.cy.spec.ts [hunk 1]
```diff
@@ -0,0 +1,152 @@
+import { SAMPLE_DATABASE } from "e2e/support/cypress_sample_database";
```
Comment (kamilmielnik) [inline e2e/test/scenarios/question-reproductions/reproductions.cy.spec.ts:1]: I started a new file for TS repros. We've lost some TS files when we merged repros into large files.

**Review by mngr — APPROVED** [review]
Works fine!

**Review by ranquild — COMMENTED** [review]
The minHeight issue exists in other (all?) places where the component is used. Unclear how to fix this at once. Maybe just add minHeight directly to `MonthLevel` in style overrides?

I don't like that we're hacking the inner styles for the uniform width. I don't know how to do it properly - maybe we shouldn't. Can we omit this from the PR? Or think how they can be implemented differently.

**Comment by kamilmielnik** [comment]
> The minHeight issue exists in other (all?) places where the component is used. Unclear how to fix this at once. Maybe just add minHeight directly to `MonthLevel` in style overrides?

Sounds good. Done: f6f46ec128de384dc7b3260f2c074ac8bc9aa480

> I don't like that we're hacking the inner styles for the uniform width. I don't know how to do it properly - maybe we shouldn't.

Right, I think it should be fixed in Mantine by adding `fullWidth` prop, but it's not supported:
- https://github.com/mantinedev/mantine/issues/958
- https://github.com/mantinedev/mantine/issues/3707

> Can we omit this from the PR?

I think it's the easiest thing to do. Done: f6f46ec128de384dc7b3260f2c074ac8bc9aa480
As a consequence we will still have this small layout shift, but it's nowhere near as annoying as the originally reported issue:

https://github.com/user-attachments/assets/fd474578-1477-403d-9bbd-e77a104b8ca5

File: frontend/src/metabase/ui/components/inputs/DatePicker/DatePicker.styled.tsx [hunk 2]
```diff
@@ -8,6 +8,7 @@ export const getDatePickerOverrides =
         /**
          * Months have different number of day rows (4, 5 or 6). This causes date picker height to change when
          * navigating between months, and the "next" & "previous" buttons will shift their positions (metabase#39487).
+         * This value should be the same as the default height of the calendar when 6 day rows are displayed.
          */
         mih: 314,
```
Comment (nemanjaglumac) [inline frontend/src/metabase/ui/components/inputs/DatePicker/DatePicker.styled.tsx:13]: Is there any way to set this value programmatically instead?
i.e. To take the height of a row and multiply it by 6. Of course, add whatever else's needed on top of that to reach 314.

Comment (kamilmielnik) [reply frontend/src/metabase/ui/components/inputs/DatePicker/DatePicker.styled.tsx:13]: It's probably an overkill :thinking:

Comment (ranquild) [reply frontend/src/metabase/ui/components/inputs/DatePicker/DatePicker.styled.tsx:13]: @kamilmielnik should it go to Calendar? Unclear, just asking

**Review by ranquild — COMMENTED** [review]
One question

File: frontend/src/metabase/ui/components/inputs/DatePicker/DatePicker.styled.tsx (hunk 2 above)
Comment (kamilmielnik) [reply frontend/src/metabase/ui/components/inputs/DatePicker/DatePicker.styled.tsx]: > @kamilmielnik should it go to Calendar? Unclear, just asking

Yeah, probably :+1: 
Done: cd0fc17f78bed9d837906bef144a03f4a29e6b25

**Review by ranquild — APPROVED** [review]

File: frontend/src/metabase/ui/components/inputs/Calendar/Calendar.styled.tsx [hunk 3]
```diff
@@ -2,6 +2,16 @@ import type { MantineThemeOverride } from "@mantine/core";
...
 
 export const getCalendarOverrides = (): MantineThemeOverride["components"] => ({
+  Calendar: {
+    defaultProps: {
+      /**
+       * Months have different number of day rows (4, 5 or 6). This causes date picker height to change when
+       * navigating between months, and the "next" & "previous" buttons will shift their positions (metabase#39487).
+       * This value should be the same as the default height of the calendar when 6 day rows are displayed.
+       */
+      mih: 314,
```
Comment (ranquild) [inline frontend/src/metabase/ui/components/inputs/Calendar/Calendar.styled.tsx:12]: nit: you could extract a constant from styles and actually compute this value

**Pull request merged**